docs/sphinx/sections/api/apidocs/external-assets.rst

```diff
@@ -0,0 +1,117 @@
+External Assets (Experimental)
+==============================
+
+Instance API
+------------
+
+External asset events can be recorded using :py:func:`DagsterInstance.report_runless_asset_event` on :py:class:`DagsterInstance`.
+
+**Example**
+
+.. code-block:: python
+
+    from dagster import DagsterInstance, AssetMaterialization, AssetKey
+
+    instance = DagsterInstance.get()
+    instance.report_runless_asset_event(AssetMaterialization(AssetKey('my_asset')))
+
+----
+
+REST API
+--------
+
+The ``dagster-webserver`` makes available endpoints for reporting asset events.
+
+/report_asset_materialization/
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A ``POST`` request made to this endpoint with the required information will result in an `AssetMaterialization` event being recorded.
+
+Parameters can be passed in multiple ways and will be considered in the following order:
+
+1. URL (``asset_key`` only)
+2. JSON body
+3. Query parameter
+
+Refer to the following table for the list of parameters and how each one can be passed to the API.
+
+**Params**
+
+.. list-table::
+   :widths: 15 15 70
+   :header-rows: 1
+
+   * - **Name**
+     - **Required/Optional**
+     - **Description**
+   * - asset_key
+     - Required
+     - **May be passed as URL path components, JSON, or a query parameter**:
+       
+       * **URL**: The asset key can be specified as path components after `/report_asset_materialization/`, where each `/` delimits parts of a multipart :py:class:`AssetKey`.
+
+       * **JSON body**: Value is passed to the :py:class:`AssetKey` constructor.
+
+       * **Query parameter**: Accepts string or JSON encoded array for multipart keys.
+   * - metadata
+     - Optional
+     - **May be passed as JSON or a query parameter**:
+     
+       * **JSON body**: Value is passed to the :py:class:`AssetMaterialization` constructor.
+
+       * **Query parameter**: Accepts JSON encoded object.
+   * - data_version
+     - Optional
+     - **May be passed as JSON or a query parameter.** Value is passed to the :py:class:`AssetMaterialization` constructor.
+   * - description
+     - Optional
+     - **May be passed as JSON or a query parameter.** Value is passed to the :py:class:`AssetMaterialization` constructor.
+   * - partition
+     - Optional
+     - **May be passed as JSON or a query parameter.** Value is passed to the :py:class:`AssetMaterialization` constructor.
+
+**Examples**
```

Comment:
I think it would be useful (and a good way to visually break things up) if we added some descriptions to these examples. Ex:

```
**curl request that <does thing>**

**curl request with JSON body**

**Python that <does thing>**
```